### ISSUE TITLE
non-0 based enums cause "result may not be initialized" warning for E.low..E.high

### ISSUE THREAD

**Issue by RedBeard0531** [nim-lang/Nim#6959]
```nim
type SomeEnum = enum
    A = 10
    B
    C
    
let slice = SomeEnum.low..SomeEnum.high
```

```
in.nim(6, 25) template/generic instantiation from here
lib/system.nim(305, 3) Warning: Cannot prove that 'result' is initialized. This will become a compile time error in the future. [ProveInit]
in.nim(6, 5) Warning: Cannot prove that 'slice' is initialized. This will become a compile time error in the future. [ProveInit]
```

Seems to happen in both stable (whatever play.nim-lang.org uses) and devel.

**Comment by twetzel59** [comment]
I have noticed this with subranges:

```nim
type
  Range = range[5..10]
var
  i = Range(5)
```

**Comment by narimiran** [comment]
Fixed in Nim 0.19.9
